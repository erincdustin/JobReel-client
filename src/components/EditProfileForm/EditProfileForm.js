import React from 'react';
import { Input, Label } from '../../components/Form/Form';
import Button from '../../components/Button/Button';

class EditProfileForm extends React.Component {
  handleSubmit = (e) => {
    e.preventDefault();
    console.log('submit');
    const firstName = e.target['first-name'].value;
    const lastName = e.target['last-name'].value;
    const username = e.target['user-name'].value;
    const email = e.target.email.value;
    const location = e.target.location.value;
    const industry = e.target.industry.value;
    const jobTitle = e.target['job-title'].value;
    const updatedUser = { id: this.props.user.id, firstName, lastName, username, email, location, industry, jobTitle }
    console.log(updatedUser);
  }

  handleClick = () => {
    console.log(`Don't submit`);
  }

  render() {
    const { user } = this.props
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <div>
            <Label htmlFor='first-name-input'>
                First Name
            </Label>
            <br />
            <Input
                name="first-name"
                id="first-name-input"
                defaultValue={user.first_name}
            />
          </div>
          <div>
            <Label htmlFor='last-name-input'>
                Last Name
            </Label>
            <br />
            <Input
                name="last-name"
                id="last-name-input"
                defaultValue={user.last_name}
            />
          </div>
          <div>
            <Label htmlFor='user-name-input'>
                User Name
            </Label>
            <br />
            <Input
                name="user-name"
                id="user-name-input"
                defaultValue={user.username}
            />
          </div>
          <div>
            <Label htmlFor='email-input'>
                Email
            </Label>
            <br />
            <Input
                name="email"
                id="email-input"
                defaultValue={user.email}
            />
          </div>
          <div>
            <Label htmlFor='location-input'>
                Location
            </Label>
            <br />
            <Input
                name="location"
                id="location-input"
                defaultValue={user.location}
            />
          </div>
          <div>
            <Label htmlFor='industry-input'>
                Industry
            </Label>
            <br />
            <Input
                name="industry"
                id="industry-input"
                defaultValue={user.industry}
            />
          </div>
          <div>
            <Label htmlFor='job-title-input'>
                Job Title
            </Label>
            <br />
            <Input
                name="job-title"
                id="job-title-input"
                defaultValue={user.job_title}
            />
          </div>
          <Button onClick={this.handleClick}>Back</Button>
          <Button type="submit">Submit</Button>
        </form>
      </div>
    )
  }
}

export default EditProfileForm;