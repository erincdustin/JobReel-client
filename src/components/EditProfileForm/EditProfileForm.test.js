import React from 'react'
import ReactDOM from 'react-dom'
import EditProfileForm from './EditProfileForm'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const dummyData = {
  first_name: 'test',
  last_name: 'test',
  username: 'test',
  email: 'test',
  city: 'test',
  industry: 'test',
  job_title: 'test'
}

describe('EditProfileForm smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<EditProfileForm user={dummyData} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('EditProfileForm snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(
        <EditProfileForm user={dummyData}/>
    )
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})