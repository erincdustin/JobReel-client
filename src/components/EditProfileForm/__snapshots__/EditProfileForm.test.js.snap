// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfileForm snapshot test renders the UI as expected 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div>
      <Label
        htmlFor="first-name-input"
      >
        First Name
      </Label>
      <br />
      <ForwardRef
        defaultValue="test"
        id="first-name-input"
        name="first-name"
      />
    </div>
    <div>
      <Label
        htmlFor="last-name-input"
      >
        Last Name
      </Label>
      <br />
      <ForwardRef
        defaultValue="test"
        id="last-name-input"
        name="last-name"
      />
    </div>
    <div>
      <Label
        htmlFor="email-input"
      >
        Email
      </Label>
      <br />
      <ForwardRef
        defaultValue="test"
        id="email-input"
        name="email"
      />
    </div>
    <div>
      <Label
        htmlFor="city-input"
      >
        City
      </Label>
      <br />
      <ForwardRef
        defaultValue="test"
        id="city-input"
        name="city"
      />
    </div>
    <div>
      <Label
        htmlFor="industry-input"
      >
        Industry
      </Label>
      <br />
      <ForwardRef
        defaultValue="test"
        id="industry-input"
        name="industry"
      />
    </div>
    <div>
      <Label
        htmlFor="job-title-input"
      >
        Job Title
      </Label>
      <br />
      <ForwardRef
        defaultValue="test"
        id="job-title-input"
        name="job-title"
      />
    </div>
    <ForwardRef
      onClick={[Function]}
      type="button"
    >
      Cancel
    </ForwardRef>
    <ForwardRef
      type="submit"
    >
      Submit
    </ForwardRef>
  </form>
</div>
`;
